test/policy-collection/policy_generator_acm_hardening.go

```diff
@@ -0,0 +1,274 @@
+		},
+		defaultTimeoutSeconds,
+		1,
+	).Should(BeTrue())
+
+	err = common.CleanupOCPUser(clientHub, clientHubDynamic, secret, user)
+	Expect(err).Should(BeNil())
+
+	err = clientHub.CoreV1().Secrets("openshift-config").Delete(context.TODO(), secret, metav1.DeleteOptions{})
+	if !k8serrors.IsNotFound(err) {
+		Expect(err).Should(BeNil())
+	}
+}
+
+var _ = Describe("GRC: [P1][Sev1][policy-grc] Test the ACM Hardening generated PolicySet in an App subscription", func() {
+	const namespace = "policies"
+	const secret = "grc-e2e-subscription-admin-user"
+	const clustersetRoleName = "grc-e2e-clusterset-role"
+	const clustersetBindingName = "grc-e2e-clusterset-binding"
+	const subAdminBinding = "open-cluster-management:subscription-admin"
+	ocpUser := common.OCPUser{
+		ClusterRoles: []types.NamespacedName{
+			{Name: "open-cluster-management:admin:local-cluster"},
+			{
+				Name:      "admin",
+				Namespace: namespace,
+			},
+		},
+		// To be considered a subscription-admin you must be part of this cluster role binding.
+		// Having the proper role in another cluster role binding does not work.
+		ClusterRoleBindings: []string{subAdminBinding, clustersetBindingName},
+		Password:            "",
+		Username:            "grc-e2e-subscription-admin",
+	}
+
+	It("Sets up the application subscription", func() {
+		By("Verifying that the subscription-admin ClusterRoleBinding exists")
+		// Occasionally, the subscription-admin ClusterRoleBinding may not exist due to some unknown
+		// error. This ClusterRoleBinding is supposed to have been created by the App Lifecycle
+		// controllers. In this unusual case, create the ClusterRoleBinding based on the advice from
+		// the Application Lifecycle squad.
+		subAdminBindingObj := rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: subAdminBinding,
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: "rbac.authorization.k8s.io",
+				Kind:     "ClusterRole",
+				Name:     subAdminBinding,
+			},
+		}
+		_, err := clientHub.RbacV1().ClusterRoleBindings().Create(
+			context.TODO(), &subAdminBindingObj, metav1.CreateOptions{},
+		)
+		if err != nil {
+			Expect(k8serrors.IsAlreadyExists(err)).Should(BeTrue())
+		}
+
+		By("Verifying that the managed cluster set binding ClusterRole exists")
+		clusterSetRule := rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: clustersetRoleName,
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups:     []string{"cluster.open-cluster-management.io"},
+					Verbs:         []string{"create"},
+					Resources:     []string{"managedclustersets/bind"},
+					ResourceNames: []string{"default"},
+				},
+			},
+		}
+
+		_, err = clientHub.RbacV1().ClusterRoles().Create(
+			context.TODO(), &clusterSetRule, metav1.CreateOptions{},
+		)
+		if err != nil {
+			Expect(k8serrors.IsAlreadyExists(err)).Should(BeTrue())
+		}
+
+		By("Verifying that the managed clusterset namespaced binding exists")
+		clustersetBinding := rbacv1.ClusterRoleBinding{
```

Comment:
You should be able to add the role in `ClusterRoles` when defining the `OCPUser` and the `ClusterRoleBinding` should be created for you.

Reply:
Isn't that only for an existing `ClusterRole`, it needs a new one to have permissions for the managed cluster set binding. @mprahl

Reply:
ah, Thanks!  Will try that soon!

Reply:
Made the update.  Thanks